config/configuration.yaml

```diff
@@ -1,12 +1,11 @@
----
 site_settings:
   site_name: localsite
   serial_number: SERIAL_123
 mqtt_settings:
   host: mosquitto
   port: 1883
   command_topic: commands/#  # We subscribe to this MQTT topic to receive commands
-  error_topic: error  # If this has a value, we publish error messages under this MQTT topic
+  error_topic: telemetry/${SITE_NAME}/error/${SERIAL_NUMBER}  # If non-empty, publish error messages under this topic
```

Comment:
You shouldn't need to "leak" this information here. The value on this configuration file should be anything. I think `error` is perfectly fine. As long as you allow to support `$ENV_VALUES` in place, this configuration would live in the other project, but you shouldn't need to make this change here in place. Not for the main config file

Reply:
OK so we should enable env var interpolation in the `error_topic` setting, but don't make it the default in this config file?

Reply:
yes
This topic should be just flexible on configuration.
But yes, the code you have written is good. Is not needed here, as part of the default configuration file.
Feel free to modify the configuration files we use in tests though

Reply:
Also, although we may not need to do the separation now, the main configuration file is in charge to do basic validations. you can move specific validations to each module.
The main configuration could be in charge only about doing YAML -> dict and ensuring can replace OS ENV values on doing that.
You can have specialised components for the input (MQTT) and the different outputs (Modbus and MQTT).
The parsing of the topic string could leave there while the environment variable remains in the main configuration file.

No change here is needed at this point. But the more complex the validations become, the more important is the question to ask, should all live together or is time to split things appart?